packages/openactive-integration-tests/test/features/payment/free-opportunities/implemented/opportunity-free-idempotency-test.js

```diff
@@ -0,0 +1,33 @@
+const { omit } = require('lodash');
+const { FeatureHelper } = require('../../../../helpers/feature-helper');
+const { FlowStageRecipes, FlowStageUtils } = require('../../../../helpers/flow-stages');
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'payment',
+  testFeature: 'free-opportunities',
+  testFeatureImplemented: true,
+  testIdentifier: 'opportunity-free-idempotency',
+  testName: 'Successful booking of free opportunity with idempotency',
+  testDescription: 'Testing idempotency of the B call for free opportunities',
+  testOpportunityCriteria: 'TestOpportunityBookableFree',
+  // This must also be TestOpportunityBookableFree as the entire Order must be free.
+  controlOpportunityCriteria: 'TestOpportunityBookableFree',
+}, (configuration, orderItemCriteriaList, featureIsImplemented, logger) => {
+  const {
+    fetchOpportunities,
+    bookRecipe,
+    defaultFlowStageParams,
+    bookRecipeArgs,
+  } = FlowStageRecipes.initialiseSimpleC1C2BookFlow(orderItemCriteriaList, logger);
+  const idempotentRepeatB = FlowStageRecipes.idempotentRepeatBAfterBook(
+    orderItemCriteriaList,
+    bookRecipe,
+    defaultFlowStageParams,
+    omit(bookRecipeArgs, ['prerequisite']),
```

Comment:
One issue was this: Idempotent B wasn't getting the same starting args as B, which was messing up its expectations for what capacity should be asserted for after the 2nd B